packages/ember-htmlbars/lib/node-managers/component-node-manager.js

```diff
@@ -122,31 +122,46 @@ function processPositionalParams(renderNode, positionalParams, params, attrs) {
   // if the component is rendered via {{component}} helper, the first
   // element of `params` is the name of the component, so we need to
   // skip that when the positional parameters are constructed
-  const paramsStartIndex = renderNode.getState().isComponentHelper ? 1 : 0;
   const isNamed = typeof positionalParams === 'string';
-  let paramsStream;
 
   if (isNamed) {
-    paramsStream = new Stream(() => {
-      return readArray(params.slice(paramsStartIndex));
-    }, 'params');
-
-    attrs[positionalParams] = paramsStream;
+    processRestPositionalParameters(renderNode, positionalParams, params, attrs);
+  } else {
+    processNamedPositionalParameters(renderNode, positionalParams, params, attrs);
   }
+}
 
-  if (isNamed) {
-    for (let i = paramsStartIndex; i < params.length; i++) {
-      let param = params[i];
-      paramsStream.addDependency(param);
-    }
-  } else {
-    for (let i = 0; i < positionalParams.length; i++) {
-      let param = params[paramsStartIndex + i];
+function processNamedPositionalParameters(renderNode, positionalParams, params, attrs) {
+  const paramsStartIndex = renderNode.getState().isComponentHelper ? 1 : 0;
+
+  for (let i = 0; i < positionalParams.length; i++) {
+    let param = params[paramsStartIndex + i];
+    if (!(positionalParams[i] in attrs)) {
```

Comment:
No, the for is iterating through the `positionalParams` array. The `paramsStartIndex` accounts for the difference if the first parameter is the name of the component (because of the `component` helper) or a direct invocation. @mixonic got rid of that in #12285 but it is a WIP